### DISCUSSION TITLE
Local File System Plugin

### DISCUSSION THREAD

**Discussion by 0xJWLabs** [wezterm/wezterm#6164] [Q&A]
### What Operating System(s) are you running on?

Windows

### WezTerm version

20240812-215703-30345b36

### Ask your question!

How to add local file system plugin in wezterm?

**Comment by alecthegeek** [comment]
Do you mean a git repo on your local file system?

Try something like this

```
local a_plugin = wezterm.plugin.require "file:///home/user/projects/myPlugin"
```

**Reply by 0xJWLabs** [reply]
wait, we can do like this? btw i wanna ask if we use git url for plugin, does wezterm download the raw files from github and save it somewhere? If yes, which folder?

**Reply by 0xJWLabs** [reply]
second question, since i also use it on my seconds pc, it uses windows, so the path should be like file://C:/..?

**Reply by alecthegeek** [reply]
> wait, we can do like this? btw i wanna ask if we use git url for plugin, does wezterm download the raw files from github and save it somewhere? If yes, which folder?

Wezterm creates a plugin folder and does a git clone into that folder. You can use the `wezterm.plugin.list()` call to discover where

> second question, since i also use it on my seconds pc, it uses windows, so the path should be like file://C:/..?

I'm not sure as I don't use Windows these days (and when I did used WSL). Looks about right...
